### PR TITLE
upgrade to node 8.14.0, warn/error if unsupported node version is used (except for Azure)

### PR DESCRIPTION
**Pull request by PieterGit**
- only allow secure Node versions from https://nodejs.org/en/blog/vulnerability/november-2018-security-releases/ (make an exception for Azure, but give warning to the user during startup)
- supported Node version is `^8.14.0`, also work with Azure Node version `8.11.1`
- error or warn if other Node versions are used
- npm update
- incorporate @sulkaharo https://github.com/nightscout/cgm-remote-monitor/pull/4129 PR for unused dependencies
- make azure deployment work with Node 8.11.1
- make sure `depcheck` won't kill swagger `/api-docs.html` 
- remove mqtt call in server.js, revert and pin socket to 2.1.1 because starting websockets takes ~  10 sec.

### CODE REVIEW

**Comment by PieterGit** [comment]
Tested with Heroku
`2018-12-11T21:10:22.523395+00:00 app[web.1]: Node v8.14.0 is recommended and supported by the Nightscout community`

Tested with Node 8.12.0 under Windows:
`Node version v8.12.0 is not supported. cgm-remote-monitor requires Node 8.14.x`

Ready for review, test by another user.

**Comment by PieterGit** [comment]
@tynbendad can you test this with Azure, and see if this PR makes deployment with Azure work out of the box. The logging should give the following WARNINGS:
```
Node 8.11.1 and Microsoft Azure is not recommended. Please migrate to another hosting provider.
Your node version is considered insecure and has several vulnerabilities. Use at your own risk.
```

**Comment by PieterGit** [comment]
Improved logic and made logging consistent.
Heroku says `Your Node version v8.14.0 is supported and recommended for Nightscout`
Test on Windows works as expected: `Node version v8.12.0 is not supported. Nightscout requires Node 8 LTS (>= 8.14.0)`
Ready for review, test and if ok merge.

**Comment by tynbendad** [comment]
summarizing from gitter: i just tested 201812_node on azure and it is working.  the log message is very hard to see, most won't see it, since azure has logs disabled by default.

File: server.js [hunk 1]
```diff
@@ -25,11 +25,40 @@
...
+// >= 8.14.0  works, supported and recommended
+// == 8.11.1  works, not fully supported (this is the latests Node version on Azure; INSECURE)
+// == 9.x     does not work, not supported
+// >= 10.14.2 works, not recommended yet, will be supported with Nightscout >= 0.12
+// >= 11.4.0  works, not recommended, will not be supported. We only support Node LTS releases
+///////////////////////////////////////////////////
+const semver = require('semver')
+var nodeVersion = process.version;
+if (semver.satisfies(nodeVersion, '^8.14.0')) { // Latest Node 8 LTS is supported and recommended for Nightscout
+  console.info('Your Node version ' + nodeVersion + ' is supported and recommended for Nightscout') ;
```
Comment (jpcunningh) [inline server.js]: suggest remove space before `;`

File: server.js [hunk 2]
```diff
@@ -25,11 +25,40 @@
...
+} else { // Not recommended Node versions. Don't start insecure versions (except 8.11.1 on Azure); Warn user of possible consequences
+  var major = semver.major(nodeVersion); // major part of the Node version number
+  var dontStart = (major === 8 && (!semver.satisfies(nodeVersion, '^8.14.0')) && (!semver.eq(nodeVersion, '8.11.1')))
+    || major<8 || major === 9 || (major === 10 && semver.lt(nodeVersion, '10.14.2'))
+    || (major === 11 && semver.lt(nodeVersion, '11.4.0'));
+  if (dontStart) { // only start on secure versions (and allow Azure version)
+    console.error('Node version '+ nodeVersion +' is not supported. Nightscout requires Node 8 LTS (>= 8.14.0)');
+    process.exit(1)
```
Comment (jpcunningh) [inline server.js]: missing `;`

File: server.js [hunk 3]
```diff
@@ -25,11 +25,40 @@
...
+  }
+  if (semver.eq(nodeVersion, '8.11.1')) {
+    console.warn('Node version v8.11.1 and Microsoft Azure are not recommended. Please migrate to another hosting provider.');
+    console.warn('Your Node version is considered insecure and has several security vulnerabilities. Use at your own risk.');
+  } else if (semver.satisfies(nodeVersion, '>=10')) {
+    console.warn('Node version ' + nodeVersion + ' is NOT recommended and may cause problems. Please use Node 8 LTS') ;
```
Comment (jpcunningh) [inline server.js]: suggest remove space before `;`

File: server.js [hunk 4]
```diff
@@ -25,11 +25,40 @@
...
+// >= 11.4.0  works, not recommended, will not be supported. We only support Node LTS releases
+///////////////////////////////////////////////////
+const semver = require('semver')
+var nodeVersion = process.version;
+if (semver.satisfies(nodeVersion, '^8.14.0')) { // Latest Node 8 LTS is supported and recommended for Nightscout
+  console.info('Your Node version ' + nodeVersion + ' is supported and recommended for Nightscout') ;
+} else { // Not recommended Node versions. Don't start insecure versions (except 8.11.1 on Azure); Warn user of possible consequences
+  var major = semver.major(nodeVersion); // major part of the Node version number
+  var dontStart = (major === 8 && (!semver.satisfies(nodeVersion, '^8.14.0')) && (!semver.eq(nodeVersion, '8.11.1')))
+    || major<8 || major === 9 || (major === 10 && semver.lt(nodeVersion, '10.14.2'))
```
Comment (jpcunningh) [inline server.js]: add spaces around `<`

File: server.js [hunk 5]
```diff
@@ -25,11 +25,40 @@
...
+    || (major === 11 && semver.lt(nodeVersion, '11.4.0'));
+  if (dontStart) { // only start on secure versions (and allow Azure version)
+    console.error('Node version '+ nodeVersion +' is not supported. Nightscout requires Node 8 LTS (>= 8.14.0)');
```
Comment (jpcunningh) [inline server.js]: add space to left of first `+` and to the right of the second `+`

**Comment by PieterGit** [comment]
Found a better way to integrate, will rewrite to `lib/server/bootevent.js` and add a unit test this weekend.

**Comment by PieterGit** [comment]
superseded by https://github.com/nightscout/cgm-remote-monitor/pull/4155

**Pull request closed without merging**